validator/client/src/main/java/tech/pegasys/artemis/validator/client/RetryingDutyLoader.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.client;
+
+import com.google.common.base.Throwables;
+import com.google.common.primitives.UnsignedLong;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.artemis.util.async.AsyncRunner;
+import tech.pegasys.artemis.util.async.SafeFuture;
+import tech.pegasys.artemis.validator.api.NodeSyncingException;
+import tech.pegasys.artemis.validator.client.duties.ScheduledDuties;
+
+public class RetryingDutyLoader implements DutyLoader {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final DutyLoader delegate;
+  private final AsyncRunner asyncRunner;
+
+  public RetryingDutyLoader(final AsyncRunner asyncRunner, final DutyLoader delegate) {
+    this.delegate = delegate;
+    this.asyncRunner = asyncRunner;
+  }
+
+  @Override
+  public SafeFuture<ScheduledDuties> loadDutiesForEpoch(final UnsignedLong epoch) {
+    final SafeFuture<ScheduledDuties> duties = new SafeFuture<>();
+    requestDuties(epoch, duties).propagateTo(duties);
```

Comment:
Any reason you can't do:
```suggestion
    return requestDuties(epoch, duties);
```

Reply:
Because we want to be able to see when the returned future is cancelled and stop retrying. If we return the one `requestDuties` provides `requestDuties` wouldn't have access to it itself to see if it's been cancelled.